src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -2458,10 +2480,7 @@ module Make_str (A : Wire_types.Concrete) = struct
                         ])
               ] )
       in
-      let current_global_slot =
-        Mina_state.Protocol_state.Body.consensus_state state_body
-        |> Consensus.Data.Consensus_state.global_slot_since_genesis_var
-      in
+      let current_global_slot = global_slot in
```

Comment:
Don't you need to make the corresponding change in `Mina_transaction_logic`? It seems like otherwise this will cause a chain halt.
(I suspect the lack of `succ` in all the tests stopped them from catching this.)

Reply:
To clarify: we use the current global slot to update the 'timing' field of accounts. For example, here, we use the `current_global_slot` from the block under construction; there, we use the global slot from the protocol state view. This will give us different values in the account, and thus an inconsistent ledger.

Likewise with `valid_until`: the notion of time in each is different, so a transaction may be applied successfully in the out-of-snark logic using the view's time, but it may fail to apply in the snark using the true global slot.

Reply:
I think I also changed the `Mina_transaction_logic.apply_transaction` to use the global_slot that I passed in. So those 2 should match.

Or did I miss any changes there?